src/main/java/teammates/ui/newcontroller/WebApiServlet.java

```diff
@@ -69,6 +69,9 @@ private void invokeServlet(HttpServletRequest req, HttpServletResponse resp) thr
         } catch (ActionMappingException e) {
             throwError(resp, e.getStatusCode(), e.getMessage());
             return;
+        } catch (UnauthorizedAccessException uae) {
```

Comment:
Rather than doing this here, we could in fact move the entire generation of the action object to the next try block.
```java
try {
    Action action = new ActionFactory().getAction(req, req.getMethod(), resp);
    action.checkAccessControl();
    ...
} catch (ActionMappingException ame) {
    throwError(resp, ame.getStatusCode(), ame.getMessage());
} catch (...) {
    ...
}
```